test/core/end2end/tests/cancel_before_invoke.cc

```diff
@@ -160,7 +160,7 @@ static void test_cancel_before_invoke(grpc_end2end_test_config config,
   error = grpc_call_start_batch(c, ops, test_ops, tag(1), nullptr);
   GPR_ASSERT(GRPC_CALL_OK == error);
 
-  cqv.Expect(tag(1), true);
+  cqv.Expect(tag(1), grpc_core::CqVerifier::AnyStatus());
```

Comment:
Instead of just accepting any result here, let's figure out what's causing this difference.  We can either try to make the behavior consistent, or at minimum we can plumb something in here to expect the right result for each of the two cases.

Reply:
done

Reply:
Instead of just accepting either response, is there an easy way we can check which stack we're using and check for the right result for that stack?  Probably not a big deal, but in principle, it seems better to be specific about what we're expecting here, so that we don't wind up with any bugs whereby the behavior of either stack becomes non-deterministic.  Also, we may not remember to update this code when we finish switching over to the promise-based stack.

Reply:
There's not... i actually tried for a while to add one, but it ends up needing an api added to grpc_call which I'm reticent to do. I've added a long-ish comment though, and we'll be coming through this code by mid next year to do the determinism pass, so I'm not too worried that we'd end up in a bad state forever.

Reply:
Okay, as long as we have a plan to deal with this, I'm happy.  Thanks!